backend/btrixcloud/migrations/migration_0039_coll_slugs.py

```diff
@@ -16,23 +21,69 @@ class Migration(BaseMigration):
     def __init__(self, mdb, **kwargs):
         super().__init__(mdb, migration_version=MIGRATION_VERSION)
 
+    async def dedup_slug(
+        self, name: str, slug_base: str, coll_id: UUID, colls_mdb
+    ) -> None:
+        """attempt to set slug, if duplicate, append suffix until a valid slug is found
+        also update original name with same suffix"""
+        slug = slug_base
+        count = 1
+
+        while True:
+            try:
+                await colls_mdb.find_one_and_update(
+                    {"_id": coll_id},
+                    {"$set": {"slug": slug}},
+                )
+                print("Slug set: " + slug)
```

Comment:
```suggestion
```

Useful for testing but will be too noisy in prod I think

Reply:
I feel like this is only printed once and helpful but could remove if you think it's in the way..

Reply:
I guess this is less helpful than the duplicate message

Reply:
I've done things like this in the past in migrations and switched to only logging errors when I found in prod we otherwise printed a huge number of lines that made tailing the backend logs to make sure the migrations didn't throw any errors more difficult.